web/src/components/vis/CorrelationTile.vue

```diff
@@ -73,6 +77,32 @@ export default {
             value: Math.abs(d.value),
           }));
     },
+    searchResults() {
+      if (!this.currentUserInput || this.currentUserInput.length === 0) {
+        return [];
+      }
+      return this.searchBarResults;
+    },
+  },
+  mounted() {
+    this.nodeCount = this.nodes.length;
+  },
+  methods: {
+    searchFilter(item, queryText, itemText) {
+      if (itemText.toLocaleLowerCase().indexOf(queryText.toLocaleLowerCase()) > -1) {
+        if (!this.searchBarResults.includes(itemText)) {
+          this.searchBarResults.push(itemText);
+        }
+        return true;
+      }
+      if (this.searchBarResults.includes(itemText)) {
+        this.searchBarResults.splice(this.searchBarResults.indexOf(itemText), 1, '');
+        if (this.searchBarResults.length > this.nodeCount) {
+          this.searchBarResults = this.searchBarResults.filter(result => result !== '');
+        }
+      }
+      return false;
```

Comment:
You've gone to some trouble to implement `Set()` here. I think this function would read much more cleanly if you just make `searchBarResults` an instance of `Set`.

That, and the added bonus that set operations are `O(1)` rather than the `O(N)` you get from `includes` and `splice`.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
To clarify, Vue doesn't have reactivity for the `Map`and `Set` data types.  You'll still have to copy values over from the set into an array, so maybe a new `searchBarResultsSet` for tracking changes, then `this.searchBarResults = Array.from(this.searchBarResultsSet)`;